doc/proposals/images-subspec.md

```diff
@@ -0,0 +1,257 @@
+      types: string
+required: ['name', 'image']
+```
+
+OpenAPIv3 Def for ImageListSpec:
+
+```yaml
+images:
+  type: array
+  items:
+    type: object
+    properties:
+      name:
+        type: string
+        description: Name of image used for lookup.
+      image:
+        type: string
+        description: Location of the image.
+      pullPolicy:
+        type: string
+        description: Pull Policy for the image.
+      pullSecrets:
+        type: array
+        description: Array of pull secret names to use.
+        items:
+          types: string
+  required: ['name', 'image']
+```
+
+#### Add an image flag to sdk add api
+
+Add a flag to preset images for operator-sdk add api command.
+
+```bash
+operator-sdk add api --api-version=app.example.com/v1alpha1 \
+  --kind=AppService --image=ngnix:latest
+operator-sdk add api --api-version=app.example.com/v1alpha1 \
+  --kind=AppService \
+  --image=ngnix:latest \
+  --image=redis:latest
+```
+
+This flag will create an AppServiceSpec struct with prefilled data.
+
+```golang
+type AppServiceSpec struct {
+
+  // Object map of image to name to an image location, pull
+  // secret names, and pull policy for the image.
+  Images operatorsdk.ImagesSpec
+
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+}
+```
+
+#### Add a generate flag for operator-sdk specs
+
+Generate flag added to operator-sdk add controller.
+
+**Blanket generate:**
+```bash
+ operator-sdk add controller --api-version=app.example.com/v1alpha1 \
+   --kind=AppService  \
+   --generate
+```
+
+The generate flag will create the controller with scaffolding using the Images
+property to create resources to be deployed. The generate command could be a
+global flag that enables any type of operatorsdk features that generate
+controller code. And we can include an option to subselect a generation.
+
+**Subselecting a generation:**
+```bash
+ operator-sdk add controller --api-version=app.example.com/v1alpha1 \
+   --kind=AppService  \
+   --generate=images
+```
+
+### Future Work
+
+#### Validation
+
+Operator score card and linters could perform validation of the use of
+ImageSpecs in the controllers.
+
+
+### Risks and Mitigations
+
+- A user could override the images in an operator with images
+  that will not work with the operator. This would just result in a broken
+  operator. The status would have to reflect a failure because of wrong software
+  version.
+- Providing default values for the images can be tricky. The default keyword in
+  the openapi v3 schema is in beta. Using enum is another method of providing a
+  default value but requires input by the creator of the CR. Mutating webhooks
+  is another method.
+
+### Upgrade / Downgrade Strategy
+
+NA.
+
+### Version Skew Strategy
```

Comment:
```suggestion
```

Reply:
We do not need to add all topics of the templates.
That ones which are not required or not make sense for the proposal can be removed/ignored.